internal/clientconn/conn.go

```diff
@@ -140,7 +146,29 @@ func (c *conn) run(ctx context.Context) (err error) {
 	}()
 
 	bufr := bufio.NewReader(c.netConn)
+
+	// if test record path is set, split netConn reader to write to file and bufr
+	if c.testRecordPath != "" {
+		if err := os.MkdirAll(c.testRecordPath, 0o644); err != nil {
+			return err
+		}
+
+		filename := fmt.Sprintf("%s_%s.log", c.netConn.RemoteAddr().String(), time.Now().Format("2006-02-01_15:04:05"))
+
+		path := filepath.Join(c.testRecordPath, filename)
+
+		//nolint:nosnakecase// os package require to use underscore
+		f, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o744)
```

Comment:
I think that should be just `os.Create`: we never append to existing file thanks to unique timestamp file name, we don't care about permissions mask beyond umask, etc.